sim/src/main.rs

```diff
@@ -36,17 +34,24 @@ fn main() {
     // Create simulator object.
     let mut simulator: Simulator = Simulator::new();
 
-    let tgen = simulator.add_node(Node::new(2, 10, 1, plugin_str, 0)); // traffic generator
-    let node1 = simulator.add_node(Node::new(2, 1, 0, plugin_str, 1));
-    let node2 = simulator.add_node(Node::new(2, 1, 0, plugin_str, 2));
-    let node3 = simulator.add_node(Node::new(2, 1, 0, plugin_str, 3));
-    let node4 = simulator.add_node(Node::new(2, 1, 0, plugin_str, 4));
-
-    let _edge5 = simulator.add_one_direction_edge(Channel::new(1, 5), tgen, node1);
-    let _edge6 = simulator.add_edge(Channel::new(2, 6), node1, node2);
-    let _edge7 = simulator.add_edge(Channel::new(2, 7), node1, node3);
-    // one way rpc sink
-    let _edge8 = simulator.add_one_direction_edge(Channel::new(1, 8), node1, node4);
+    // node arguments go:  id, capacity, egress_rate, generation_rate, plugin
+    simulator.add_node("traffic generator".to_string(), 10, 1, 1, plugin_str);
```

Comment:
Cool, this looks pretty good already. One thing about the `to_string()` methods. If you declare the type to be &str you should not need to do this every time. In general, we should take a closer look on how we handle strings in this simulator (and the compiler, too). The fact that we have to use `clone()` and `to_string() ` **a lot**, indicates we are doing something wrong. These operations are a) inefficient and b) hard to read. They should only be used sparingly, i.e., only when we actually modify that particular string. Often, that is not the case.